ssl/statem/statem_lib.c

```diff
@@ -1665,9 +1665,9 @@ static void check_for_downgrade(SSL *s, int vers, DOWNGRADE *dgrd)
     if (vers == TLS1_2_VERSION
             && ssl_version_supported(s, TLS1_3_VERSION, NULL)) {
         *dgrd = DOWNGRADE_TO_1_2;
-    } else if (!SSL_IS_DTLS(s) && vers < TLS1_2_VERSION
-            && (ssl_version_supported(s, TLS1_2_VERSION, NULL)
-                || ssl_version_supported(s, TLS1_3_VERSION, NULL))) {
+    } else if (!SSL_IS_DTLS(s)
+            && vers < TLS1_2_VERSION
+            && ssl_version_supported(s, TLS1_2_VERSION, NULL)) {
```

Comment:
Perhaps another place for a comment clarifying the intent.  I think here we're making sure that a server that disables TLS 1.2, creating a hole between 1.1 and 1.3 can still complete handshakes with clients that support 1.2 and below. If we also tested for 1.3 support, we'd signal a 1.1 downgrade and the client would fail, right?  This looks like the scenario David Benjamin was describing... If so, the change makes sense, but substantive comments are needed to highlight a non-trivial decision made here...